gammapy/cube/core.py

```diff
@@ -312,21 +312,15 @@ def to_sherpa_data3d(self):
     @property
     def solid_angle_image(self):
         """Solid angle image in steradian (`~astropy.units.Quantity`)"""
-        cube_hdu = fits.ImageHDU(self.data, self.wcs.to_header())
+        skymap = self.sky_image()
+        return skymap.solid_angle()
 
-        from .utils import cube_to_image
-        image_hdu = cube_to_image(cube_hdu)
-        image_hdu.header['WCSAXES'] = 2
-
-        sky_map = SkyMap.from_image_hdu(image_hdu)
-        return sky_map.solid_angle()
-
-    def sky_image(self, idx_energy, copy=True):
+    def sky_image(self, idx_energy=None, copy=True):
         """Slice 3-dim cube into a 2-dim `~gammapy.image.SkyMap` along slicepos.
 
         Parameters
         ----------
-        inx_energy : int
+        idx_energy : int or None (default None)
```

Comment:
> Are you sure that getting the first image is the same as summing over the first axis?

Well, the values in the `data` array are going to be different. In one case it'll be the sum along the energy axis, in the other the values from the first energy slice.

But for `SkyImage.solid_angle` it doesn't matter at all what the values in the `SkyImage.data` array are.
So yes, for this application it will be the same.

Makes sense?

As I said, if we do want to add convenience methods to combine bins along the energy axis, I would prefer a separate method (and also different methods how to combine the values).